### ISSUE TITLE
Low Level API with Pre Allocated Work Space Exposed

### ISSUE THREAD

**Issue by RoyiAvital** [sarah-ek/gemm#24]
It would be great if there will be, for any function, a low level API which exposes the needed workspace to avoid any allocations of the function.

The work is impressive. Being competitive with the big guys is nothing short of amazing considering this is a single person show.

**Comment by donkeyteethUX** [comment]
Agreed. My application calls faer's `apply_block_householder_sequence_transpose_on_the_left_in_place_with_conj` which already accepts a `stack: &mut MemStack` parameter for scratch space, so I was surprised when profiling showed that it still allocated internally, inside `gemm_basic_generic`. It seems like the user-provided scratch buffer could be passed down to `gemm` to avoid this.

Also, thanks so much for your work on the faer/dynstack/gemm ecosystem. It's fantastic.
